src/js/editor/post.js

```diff
@@ -602,20 +602,23 @@ class PostEditor {
    * or, if a string, the tag name of the markup (e.g. 'strong', 'em') to toggle.
    */
   toggleMarkup(markupOrMarkupString) {
-    const markup = typeof markupOrMarkupString === 'string' ?
-                     this.builder.createMarkup(markupOrMarkupString) :
-                     markupOrMarkupString;
-
-    const range = this.editor.cursor.offsets;
-    const hasMarkup = m => m.hasTag(markup.tagName);
-    const rangeHasMarkup = any(this.editor.markupsInSelection, hasMarkup);
-
-    if (rangeHasMarkup) {
-      this.removeMarkupFromRange(range, hasMarkup);
-    } else {
-      this.applyMarkupToRange(range, markup);
+    if (this.editor.cursor.hasSelection()) {
```

Comment:
I can- but I'm not sure this will ever run without a selection. Without a selection we would do something else like run the stateful code.

And avoiding the `selectRange` when it is not needed (when it is not broken) seems ideal.